src/cpp/rtps/builtin/discovery/participant/PDP.cpp

```diff
@@ -1060,13 +1060,8 @@ bool PDP::remove_remote_participant(
             std::lock_guard<std::mutex> lock(callback_mtx_);
             ParticipantDiscoveryInfo info(*pdata);
             info.status = reason;
-            bool should_ignore;
             listener->onParticipantDiscovery(mp_RTPSParticipant->getUserRTPSParticipant(), std::move(
-                        info), should_ignore);
-            if (should_ignore && !mp_RTPSParticipant->is_participant_ignored(pdata->m_guid.guidPrefix))
-            {
-                mp_RTPSParticipant->ignore_participant(pdata->m_guid.guidPrefix);
-            }
+                        info));
```

Comment:
I think we should still call the new version of the callback, since the idea is to mark the older version as deprecated.
We will document that `should_ignore` is only taken into account for certain info.status values (DISCOVERED_PARTICIPANT and CHANGED_QOS_PARTICIPANT)